doc/concept/build_system.md

```diff
@@ -0,0 +1,73 @@
+
+**Fully internal dependencies:** (ABK) Built an ad-hoc package manager (thin wrapper on top of `git clone`) into the project's configure stage so that the project can compile on any computer for any platform (within reason).  This includes required 3rd party libraries, required 3rd party tools, you name it.  Setting up a developer machine no longer requires any special work past SSH keys, a working compiler and such.
+
+
+Bad experiences to avoid
+------------------------
+
+**Boost version difference broke build:** (ABK) While onboarding a new developer, a difference in a Boost version caused an obscure build failure.  Takeaway: when practical, always freeze dependencies using some sort of package manager.  (Is this takeaway actually practical in an open-source project where the dependencies are almost by definition non-frozen?)
+
+
+Available build systems
+-----------------------
+
+| Name | Namesake | Overview | Notable Advantages | Notable Disadvantages |
+| ---- | -------- | -------- | ------------------ | --------------------- |
+| [A-A-P](https://en.wikipedia.org/wiki/A-A-P)                  | | Python-based package manager written by the author of vim. | Augmentable via scripting (python), supports uploading and downloading files | Not designed for building software
+| [Ant](https://en.wikipedia.org/wiki/Apache_Ant)               | "Another Neat Tool" | Java+XML-based build system | Widely used, straight-forward, relatively easy to read if you don't know Ant. | Build specification file is not as powerful as other options here; extensions required if you want to improve things.<br />Non-Java languages feel second-class<br />Ant extensions are developed using a different SDK than the main build specification file.<br />
+| [Autoconf](https://en.wikipedia.org/wiki/Autoconf) & [Automake](https://en.wikipedia.org/wiki/Automake) | auto wrappers on top of make | Automates the creation of Makefiles for C-based projects | Widely used | Complexity, difficult to debug
+| [Bazel](https://en.wikipedia.org/wiki/Bazel_(software))       | Anagram of Blaze, Google's internal build system | Java-based Build system using Skylark (flavor of Python)-based configuration files for describing the build | Parallel by default, supported by Google, (computer) language-agnostic. Repeatable, scalable builds<br />Xcode support: bypasses Xcode to an extent; builds various Apple components manually
+| [Buck](https://en.wikipedia.org/wiki/Buck_(software))         | Facebook's internal build system | | Native integration with Xcode and IntelliJ | Native integration in only Xcode and IntelliJ
+| [BuildAMation](https://en.wikipedia.org/wiki/BuildAMation)    | | A modular build system using C#-like syntax
+| [Buildr](https://en.wikipedia.org/wiki/Apache_Buildr)         | | A Ruby-based competitor to Ant. | | | Intended primarily for Java projects
+| [CMake](https://en.wikipedia.org/wiki/CMake)                  | | A cross-platform, modular wrapper around Make
```

Comment:
My preference is this (CMake) would be the primary supported build system.

Reply:
FWIW, I agree. I don't have a preference of a secondary build system.